src/openMVG/sfm/pipelines/sequential/sequential_SfM.cpp

```diff
@@ -394,6 +396,38 @@ bool SequentialSfMReconstructionEngine::Process()
     // Add process time
     _tree.put("sfm.time", time_sfm);
 
+    #ifdef __linux__
+      // Add CPU/RAM infos
+      char values[20];
+      std::ifstream ifs;
+
+      if(!std::system("nproc >values.txt"))
```

Comment:
This is dangerous since it because it has the potential to destroy user's data: it will unconditionally overwrite the file `values.txt` in the program's current directory. There are multiple ways of fixing this:

1. Open `/proc/cpuinfo` and `/proc/meminfo`  directly and parse the contents manually
2. Use `mkstemp` to generate a unique temporary filename.
3. Use `tmpfile` to create and open a unique temporary file.
4. Use `popen` to open a pipe to the process and avoid creation of temporary.